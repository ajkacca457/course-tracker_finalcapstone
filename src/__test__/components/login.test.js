import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import Login from '../../components/login';
import store from '../../redux/store';

describe('Home Component', () => {
  it('renders correctly', () => {
    const hometree = TestRenderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Login />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(hometree).toMatchSnapshot();
  });

  test('renders the title text of the Login page', () => {
    const { getByText } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>,
    );
    const title = getByText('Login with email and password');
    expect(title).toBeInTheDocument();
  });

  test('renders button of Login page', () => {
    const { getByText } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>,
    );
    const button = getByText('Login');
    expect(button).toBeInTheDocument();
  });

  test('renders button of Login page', () => {
    const { getByText } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>,
    );
    const button = getByText('Signup');
    expect(button).toBeInTheDocument();
  });
});
