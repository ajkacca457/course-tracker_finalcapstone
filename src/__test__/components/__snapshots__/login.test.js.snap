// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders correctly 1`] = `
<div
  className="logincont"
>
  <div
    className="bg-white mx-auto p-2 rounded loginitem"
  >
    <h5
      className="text-center bg-info py-2 text-white mb-3 rounded"
    >
      Login with email and password
    </h5>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group text-left"
      >
        <input
          className="form-control"
          id="username"
          onChange={[Function]}
          placeholder="Enter username"
          type="text"
        />
      </div>
      <div
        className="form-group text-left"
      >
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </div>
      <button
        className="btn btn-info w-100"
        type="submit"
      >
        Login
      </button>
    </form>
    <div
      className="signupinfo text-right"
    >
      <p
        className="tagline"
      >
        Dont have an account?
      </p>
      <a
        className="nav-link m-0 bg-danger w-25 ml-auto text-center text-white tagbtn"
        href="/signup"
        onClick={[Function]}
      >
        Signup
      </a>
    </div>
  </div>
</div>
`;
